ksql-execution/src/main/java/io/confluent/ksql/execution/util/EngineProcessingLogMessageFactory.java

```diff
@@ -35,7 +35,7 @@
   private EngineProcessingLogMessageFactory() {
   }
 
-  public static Function<ProcessingLogConfig, SchemaAndValue> recordProcessingError(
+  public static synchronized Function<ProcessingLogConfig, SchemaAndValue> recordProcessingError(
```

Comment:
why are we making this `synchronized`? It probably doesn't have any effect since we're returning a lambda anyway (this method doesn't access any state - if you want to synchronize the inside of the lambda that needs to be done separately)

Reply:
You are right that it was placed in the wrong place. I thought I needed to synchronize the writing to the logger but it turns out it is multi-threaded after talking with Rohan.